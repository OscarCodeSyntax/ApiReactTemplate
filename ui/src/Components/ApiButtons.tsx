import * as React from "react";
import BasicQueries from "../Resources/DataService/BasicQueries";
import { useState } from "react";

const ApiButtons = () => {

  const [screenValue, setscreenValue] = useState<String>("Null");

  async function deleteOnClick()  {
   if(await BasicQueries.deleteById(2)){
    setscreenValue("Deleted")
   }
  }

  async function findAllOnClick () {
    if(await BasicQueries.deleteById(2)) {
    setscreenValue("Found All")
    }
  }

  async function updateOnClick () {
    if(await BasicQueries.update(2, "updated")) {
    setscreenValue("Updated")
    }
  }


  
  return (
    <>
    {screenValue}
      <div>
        <button
          onClick={(e: any) => BasicQueries.create("create").then(setscreenValue)}
          title="Create"
          value= "Create"
        >Create </button>
        <button
          onClick={(e: any) => {
            deleteOnClick()
          }}
          title="Delete By Id"
          value="Delete By Id"
        >Delete By Id </button>
        <button
          onClick={(e: any) => {
            findAllOnClick()
          }}
          title="Find All"
          value="Find All"
        >Find All </button>
        <button
          onClick={(e: any) => {
            BasicQueries.findById(1).then(setscreenValue);
          }}
          title="Find By Id"
          value="Find By Id"
        >Find By Id </button>
        <button
          onClick={(e: any) => {
            updateOnClick()
          }}
          title="Update"
          value="Update"
        >Update </button>
      </div>
      <div></div>
    </>
  );
};

export default ApiButtons;
