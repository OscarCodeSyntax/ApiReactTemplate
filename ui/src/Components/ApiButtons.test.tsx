import React from "react";
import ApiButtons from "./ApiButtons";
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import BasicQueries from "../Resources/DataService/BasicQueries";



describe('Api Buttons ', () => {

 beforeEach(()=>{

    jest.resetAllMocks();
 })

 it('renders correctly', () => {
    const tree = renderer
    .create(<ApiButtons />)
    .toJSON();
  expect(tree).toMatchSnapshot();
 });

//  it('renders correct buttons with text', () => {

//     const  api= shallow(<ApiButtons />);

//  }
//  )

})