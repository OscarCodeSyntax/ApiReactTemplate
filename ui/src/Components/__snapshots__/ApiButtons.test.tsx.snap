// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api Buttons  renders correctly 1`] = `
[
  "Null",
  <div>
    <button
      onClick={[Function]}
      title="Create"
      value="Create"
    >
      Create 
    </button>
    <button
      onClick={[Function]}
      title="Delete By Id"
      value="Delete By Id"
    >
      Delete By Id 
    </button>
    <button
      onClick={[Function]}
      title="Find All"
      value="Find All"
    >
      Find All 
    </button>
    <button
      onClick={[Function]}
      title="Find By Id"
      value="Find By Id"
    >
      Find By Id 
    </button>
    <button
      onClick={[Function]}
      title="Update"
      value="Update"
    >
      Update 
    </button>
  </div>,
  <div />,
]
`;
